src/pip/_internal/models/wheel.py

```diff
@@ -32,20 +39,40 @@ def __init__(self, filename: str) -> None:
         self.name = wheel_info.group("name").replace("_", "-")
         _version = wheel_info.group("ver")
         if "_" in _version:
-            deprecated(
-                reason=(
-                    f"Wheel filename {filename!r} uses an invalid filename format, "
-                    f"as the version part {_version!r} is not correctly normalised, "
-                    "and contains an underscore character. Future versions of pip may "
-                    "fail to recognise this wheel."
-                ),
-                replacement=(
-                    "rename the wheel to use a correctly normalised version part "
-                    "(this may require updating the version in the project metadata)"
-                ),
-                gone_in="25.1",
-                issue=12914,
-            )
+            try:
+                parse_wheel_filename(filename)
+            except InvalidVersion as e:
+                deprecated(
+                    reason=(
+                        f"Wheel filename version part {_version!r} is not correctly "
+                        "normalised, and contained an underscore character in the "
+                        "version part. Future versions of pip will fail to recognise "
+                        f"this wheel and report the error: {e.args[0]}."
+                    ),
+                    replacement=(
+                        "rename the wheel to use a correctly normalised "
+                        "version part (this may require updating the version "
+                        "in the project metadata)"
+                    ),
+                    gone_in="25.1",
+                    issue=12938,
+                )
+            except PackagingInvalidWheelName as e:
+                deprecated(
+                    reason=(
+                        f"The wheel filename {filename!r} is not correctly normalised. "
+                        "Future versions of pip will fail to recognise this wheel. "
+                        f"and report the error: {e.args[0]}."
```

Comment:
This reads poorly:

```
DEPRECATION: The wheel filename 'six-1.16.0_build1-py3-none-any.whl' is not correctly normalised. Future versions of pip will fail to recognise this wheel. and report the error: Invalid wheel filename (invalid version): six-1.16.0_build1-py3-none-any. pip 25.1 will enforce this behaviour change. A possible replacement is rename the wheel to use a correctly normalised name (this may require updating the version in the project metadata). Discussion can be found at https://github.com/pypa/pip/issues/12938
ERROR: Invalid requirement: 'six==1.16.0-build1': Expected end or semicolon (after version specifier)
    six==1.16.0-build1
       ~~~~~~~~~~^
```

That period seems superfluous? "pip will fail to recognise this wheel. and report the error"

Reply:
😞 , yes, I should have done another read over of the error before pushing the commit, I didn't expect it to actually ever trigger though.

Does your example actually cause this error to trigger? i.e. it passes the regex but not the `parse_wheel_filename`

I can make yet another follow up to improve the wording.

Reply:
I made it up by renaming the random six wheel I have lying around. It should not be considered a real world example :P

Reply:
I'll make a PR tomorrow aimed to be as small as possible to clean up the wording. Thanks for pointing out.